ENIGMAsystem/SHELL/Audio_Systems/SDL/SDLsystem.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include <map>
+#include <vector>
+#include <SDL.h>
+#include "Widget_Systems/widgets_mandatory.h"
+#include "SDLsystem.h"
+#include "SoundResource.h"
+
+AssetArray<Sound> sounds;
+namespace enigma {
+int audiosystem_initialize() {
+  SDL_Init(SDL_INIT_AUDIO);
+  int flags=MIX_INIT_OGG|MIX_INIT_MOD|MIX_INIT_FLAC|MIX_INIT_MP3;
+  int initted=Mix_Init(flags);
+  if ((initted&flags) != flags) {
+    DEBUG_MESSAGE(std::string("Mix_Init: %s\n")+ std::string(Mix_GetError()),MESSAGE_TYPE::M_ERROR);

```

Comment:
Please do not use `Mix_GetError()` and either use a generic "failed to initialize" or find a better method. I noticed it being unreliable because it was telling me irrelevant things like the window wasn't focused, which is totally bogus. It's also mentioned in the doc that it's useless and should NOT be relied upon.

https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_9.html#SEC9
>"Note: this function does not always set the error string, so do not depend on Mix_GetError being meaningful all the time."

Reply:
resolved